src/compiler/checker.ts

```diff
@@ -35117,11 +35117,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const len = signature.parameters.length - (signatureHasRestParameter(signature) ? 1 : 0);
         for (let i = 0; i < len; i++) {
             const declaration = signature.parameters[i].valueDeclaration as ParameterDeclaration;
-            if (declaration.type) {

```

Comment:
I noticed that this check is somewhat redundant because `getEffectiveTypeAnnotationNode` already resolves this and that it actually prevents from this function to work on functions annotated using JSDoc